File: Data Structures & Algorithms/two-integer-sum/submission-0.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        var m = new Dictionary<int, int>();

        for(var i = 0; i < nums.Length; i ++)
        {
            if (m.TryGetValue(target - nums[i], out int i2))
                return [i2, i];
            
            m[nums[i]] = i;
        }

        return [];
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.go
func twoSum(numbers []int, target int) []int {
    begin := 0
    end := len(numbers) - 1

    for begin < end {
        sum := numbers[begin] + numbers[end]
        if sum == target {
            return []int{ begin +1, end +1}
        } else if sum < target {
            begin++
        } else {
            end--
        }
    }

    return []int{}
}
